source/fab/language/fortran/reader.py

```diff
@@ -0,0 +1,43 @@
+# (c) Crown copyright Met Office. All rights reserved.
+# For further details please refer to the file COPYRIGHT
+# which you should have received as part of this distribution
+
+'''
+Source reading tools for Fortran format.
+'''
+import re
+
+
+def sourcefile_iter(filename: str) -> str:
```

Comment:
As above, thus function yields a generator of strings, not strings themselves.

Are you aware of `pathlib` which has been added to Python 3 and which improves handling of file paths? Passing `filename` as a string is fine for the moment but it's worth looking into for the future.

Reply:
I'll see what I can do (it'd be good to avoid putting off using something more suitable especially when it's such a small set of code to update)

Reply:
Hopefully that seems okay?  `pathlib` is certainly a lot neater